client/src/screens/training/view.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+
+import { Button, Layout, NavbarContainer } from '../../components';
+import { connect, requireData } from '../../utils';
+import { controller } from './controller';
+import { ReactComponent as EditIcon } from '../../assets/edit.svg';
+import { ReactComponent as BackIcon } from '../../assets/backArrow.svg';
+
+import style from './style.module.scss';
+
+class PureTraining extends React.Component {
+  renderTopBar = () => {
+    const { data } = this.props;
+    return(
+      <div className={style.topBar}>
+        {data.name}
+      </div>
+    );
+  };
+
+  renderBottomBar = () => {
+    const { onBack, onEdit } = this.props;
+    return (
+      <NavbarContainer className={style.navbarContainer}>
+        <Button
+          skin="icon"
+          size="large"
+          onClick={onBack}
+        >
+          <BackIcon />
+        </Button>
+        <Button
+          skin="icon"
+          size="large"
+          onClick={onEdit}
+        >
+          <EditIcon />
+        </Button>
+      </NavbarContainer>
+    );
+  };
+
+  renderExercises = () => {
+    const { data, onOpenExercise } = this.props;
+    let sets = '';
+    return (
+      <ul className={style.exercises}>
+        {data.exercises.map(exercise => {
+          return (
+            <li
+              key={exercise.id}
+              className={style.exercise}
+            >
+              <div 
+                className={style.exerciseBox}
+                onClick={onOpenExercise}  
+              >
+                {exercise.name}
+                <br/>
+                {exercise.sets.map((set, index) => {
+                  sets += `${set.repetitions}x${set.weight}kg`;
+                  if (index < (exercise.sets.length - 1)) sets += ' - ';
+                })}
+                {sets}
+                {sets = ''}
```

Comment:
disgusting!
all such calculations should be done in the controller.
only ready values should be passed in UI.